controllers/snapshot_controller.go

```diff
@@ -58,8 +59,22 @@ func (r *SnapshotReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, fmt.Errorf("failed to create patch helper: %w", err)
 	}
 
-	obj.Status.Image = fmt.Sprintf("http://%s/snapshots/%s", r.RegistryServiceName, obj.Spec.Ref)
-	obj.Status.Layer = fmt.Sprintf("http://%s/snapshots/%s@%s", r.RegistryServiceName, obj.Spec.Ref, obj.Spec.Digest)
+	//repoName := fmt.Sprintf("%s/%s", r.RegistryServiceName, obj.Spec.Ref)
+	repo, err := oci.NewRepository(r.RegistryServiceName + "/")
+	if err != nil {
+		return ctrl.Result{}, fmt.Errorf("failed to create/find repo with name '%s': %w", r.RegistryServiceName, err)
+	}
+	// todo: filters provide extra identity search
+	manifest, _, err := repo.FetchManifest(obj.Spec.Ref, nil)
```

Comment:
as it is `FetchManifest` needs the whole path. We need to adapt it to do completion with repo name

Reply:
FetchManifest already does `parseReference` that concatenates the two together ( the repository and the reference ).

```go
	ref, err := parseReference(reference, r)
	if err != nil {
		return nil, nil, fmt.Errorf("failed to parse reference: %w", err)
	}
```

So If I pass in the whole path it will result in the duplicated path from the repo's name + `:` + the ref. So we either need to be consistently do that or never do it. :)

Reply:
yes I have been tired yesterday. There is nothing better to do here.